webapp/src/ts/services/usage-analytics.service.ts

```diff
@@ -0,0 +1,171 @@
+    if (!isConfigOkay) {
+      return;
+    }
+
+    this.loadScript();
+  }
+
+  private async setConfig() {
+    const settings = await this.settingsService.get().catch(() => null);
+    const siteId = settings?.usage_analytics?.site_id;
+    this.analyticsServer = settings?.usage_analytics?.server_url;
+
+    if (!this.analyticsServer || !siteId) {
+      console.warn(`Usage Analytics :: Missing configuration. Server URL: ${this.analyticsServer} Site ID: ${siteId}`);
+      return false;
+    }
+
+    this.window._paq.push([ MatomoConfig.DISABLE_CAPTURE_KEYSTROKES ]);
+    this.window._paq.push([ MatomoConfig.MATCH_TRACKER_URL ]);
+    this.window._paq.push([ MatomoConfig.TRACK_PAGE_VIEW ]);
+    this.window._paq.push([ MatomoConfig.ENABLE_LINK_TRACKING ]);
+    this.window._paq.push([ MatomoConfig.SET_TRACKER_URL, `${this.analyticsServer}/${this.THIRD_PARTY_TRACKER_FILE}` ]);
+    this.window._paq.push([ MatomoConfig.SET_SITE_ID, siteId.toString() ]);
+
+    return true;
+  }
+
+  private async canTrack() {
+    return !this.sessionService.isDbAdmin() && await this.authService.has(CAN_TRACK_USAGE_ANALYTICS);
+  }
+
+  private loadScript() {
+    const head = this.document.getElementsByTagName('head')[0];
+    const script = this.document.createElement('script');
+
+    script.type = 'text/javascript';
+    script.async = true;
+    script.src = `${this.analyticsServer}/${this.THIRD_PARTY_SCRIPT_FILE}`;
+
+    this.isScriptReady.subscribe(isReady => isReady && this.startTracking());
+    script.onload = () => this.isScriptReady.next(true);
+
+    head.appendChild(script);
+  }
+
+  private startTracking() {
+    console.info('Usage analytics tracking started. Support compatible with ' +
+      this.THIRD_PARTY_SCRIPT_FILE + ' version ' + this.THIRD_PARTY_SCRIPT_VERSION);
+
+    this.router.events
+      .pipe(filter(event => event instanceof NavigationEnd))
+      .subscribe(() => {
+        this.ngZone.runOutsideAngular(() => this.trackNavigation());
+      });
+  }
+
+  private trackNavigation() {
+    const module = this.getModuleFromCurrentUrl();
+    if (module) {
+      this.window._paq.push([ MatomoConfig.SET_DOCUMENT_TITLE, module ]);
+    }
+
+    if (this.previousPageUrl) {
+      this.window._paq.push([ MatomoConfig.SET_REFERENCE_URL, this.previousPageUrl ]);
+    }
+
+    const currentUrl = this.replaceUuid(this.window.location.href);
+    this.window._paq.push([ MatomoConfig.SET_CUSTOM_URL, currentUrl ]);
+    this.previousPageUrl = currentUrl;
+
+    this.window._paq.push([ MatomoConfig.TRACK_PAGE_VIEW ]); // Set last
+    this.window._paq.push([ MatomoConfig.ENABLE_LINK_TRACKING ]); // Also, set last
+  }
+
+  private replaceUuid(url) {
+    if (!this.UUID_REGEX.test(url)) {
+      return url;
+    }
+
+    const newURL = url.replace(this.UUID_REGEX, 'UUID');
+    const navigation = this.router.getCurrentNavigation();
+    const usageAnalyticsValue = navigation?.extras?.state?.usageAnalyticsValue;
```

Comment:
We don't need the UUIDs, instead to understand the reports, we need the contact_type and the form name